test/minitest/test_minitest_mock.rb

```diff
@@ -495,4 +495,18 @@ def test_mock_with_yield
     @tc.assert_equal true, rs
   end
 
+  def test_blow_up_on_multiple_stub
+    h = {}
+    e = assert_raises StubError do
+      h.stub(:empty?, false) do
+        h.stub(:empty?, true) do
```

Comment:
Okay I wasn't sure. Yeah I'm happy to drop it since it is not gonna make a big improvement on Minitest. I would rather do something which is useful.

Although "rare enough" would be a better reason for me than "bad enough". Using the wrong number of arguments on a method is "bad", but unfortunately "ArgumentError: wrong number of arguments" is not that rare. Exceptions rarely show when your code is "good", that is the point, isn't it?

I reckon it is rare enough to drop the pull request.